nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/test/java/org/apache/nifi/cluster/manager/NarSummariesMergerTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one or more
+ *  contributor license agreements.  See the NOTICE file distributed with
+ *  this work for additional information regarding copyright ownership.
+ *  The ASF licenses this file to You under the Apache License, Version 2.0
+ *  (the "License"); you may not use this file except in compliance with
+ *  the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.nifi.cluster.manager;
+
+import org.apache.nifi.cluster.protocol.NodeIdentifier;
+import org.apache.nifi.web.api.dto.NarCoordinateDTO;
+import org.apache.nifi.web.api.dto.NarSummaryDTO;
+import org.apache.nifi.web.api.entity.NarSummaryEntity;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class NarSummariesMergerTest {
+
+    @Test
+    public void testMergeNarSummaries() {
+        final NarSummaryEntity narSummary1 = createNarSummaryEntity("org.apache.nifi", "nifi-nar-1", "1.0.0");
+        final NarSummaryEntity narSummary2 = createNarSummaryEntity("org.apache.nifi", "nifi-nar-2", "1.0.0");
+        final NarSummaryEntity narSummary3 = createNarSummaryEntity("org.apache.nifi", "nifi-nar-3", "1.0.0");
+        final NarSummaryEntity narSummary4 = createNarSummaryEntity("org.apache.nifi", "nifi-nar-4", "1.0.0");
```

Comment:
Recommend defining and reusing host, version, and port variables across this test method.